src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +170,264 @@ public void recordTagValue(String time)
+    *
+    * @return Returns a string of the file contents. it will be the epoch time pointer for the data queue.
+    * also called a timeInt in the return file from the EBD commands. name will be timeInt throughout code.
+    */
+   public String readTimePointerFile()
+   {
+      String timeInt = "-1";
+      // check if file exists
+      File tempFile = new File(tagTimeIntFileName);
+      boolean exists = tempFile.exists();
+
+      if(exists) {
+         BufferedReader reader;
+         try {
+            reader = new BufferedReader(new FileReader(tagTimeIntFileName));
+            String line = reader.readLine();
+            while (line != null) {
+
+               // there will be just one line, but just save last line.
+               timeInt = line;
+               // Logger.LOG_INFO("line from time file is: "+line);
+               line = reader.readLine();
+            }
+
+         } catch (IOException e) {
+             e.printStackTrace();
+             Logger.LOG_ERR("Something went wrong reading tag data queue time pointer from timeInt file. Check tag timeInt file: "+tagTimeIntFileName);
+         }
+
+      } else { // when connector has not run before, these files will not exist.
+
+          // if file does not exist, set timeInt to the current time to start a new queue
+          Logger.LOG_INFO(getTagName()+": time int was null, setting back of tag data queue current time");
+          long tmpTimeInt = System.currentTimeMillis();
+          // current time (when the list was null) will be the new back of queue
+          String writtenValue = writeTimePointerFile(tmpTimeInt);
+          // return what got written in the writeTimePointerFile function
+          return writtenValue;
+      }
+
+      // return the string read in this function
+      return timeInt;
+   }
+
+   /**
+    * This function generates a list of historical logs from
+    * and EBD call and then reads the list into the internal queue
+    *
+    */
+   public void readFromHistoricalLogQueue()
+   {
+       // EBD call to generate tag data point file
+       exportBlockDescriptor();
+
+       // parse the file generated by the EBD call and add all info to array list
+       parseTagFile();
+   }
+
+   /**
+    * Takes in timeInt retrieved from EBD call to historical logs and converts it to the format
+    * needed for OSIsoft.
+    *
+    * @param timeInt time from historical logs which is in epoch time
+    * @return returns the new string to send to OSIsoft
+    */
+   private String convertTimeIntHelper(long timeInt)
+   {
+
+      String formattedTime = "";
+      formattedTime = new SimpleDateFormat("ddMMyyyy_HHmmss")
+               .format(new Date(timeInt));
+      return formattedTime;
+   }
+
+   /**
+    *  remove from the queue by moving the time pointer forward to a later time
+    * @param numTags number of places to move forward in the queue
+    */
+   public void popNumTagsFromQueue(int numTags)
+   {
+      if (numTags != 0) {

```

Comment:
what if numTags = -1?

Reply:
I mean, I'll change that to greater than 0 but im pretty sure it is impossible. numTags will only be the pointsAdded from the DataPoster class which is 0 or greater

Reply:
Just defensive coding.